evasquez26/angular4-startup
# S2 ONE 32 Password reset

## Story / Task

As a user I can reset my password
see [ONE-32](https://bixlabs.atlassian.net/browse/ONE-32)


## What this PR does?

Adds functionality to reset your password

### Before 

There was no functionality to reset one's password

### After

We can ask to reset our password
We can reset our password by using the email link

### How to test / reproduce

- Click sign in button in header
- Click "Forgot password" link
- Fill out email and submit.
- Verify you recieve an email
- Click the link in the email (Actual link in email should be updated as it points to `1neschool.com` so you can go to `localhost:4200/recover-password?token=TOKEN_FROM_EMAIL` instead)
- Fill out the form and submit
- You sould be able to see a success message and login with new password (Currently it responds with `Unauthorized`. Will ask Lev about this)

### TODOs
- [x] Verify update credentials endpoint

### Related PRs
N/A

## Migrations

NO


Thanks, @evasquez26
